src/util/taproot.rs

```diff
@@ -700,8 +700,7 @@ impl ControlBlock {
 
     /// Serialize to a writer. Returns the number of bytes written
     pub fn encode<Write: io::Write>(&self, mut writer: Write) -> io::Result<usize> {
-        let first_byte: u8 =
-            (if self.output_key_parity { 1 } else { 0 }) | self.leaf_version.as_u8();
+        let first_byte: u8 = i32::from(self.output_key_parity) as u8 | self.leaf_version.as_u8();
```

Comment:
Here, we rely on Parity being only one bit. But converting to u8 should be enough.